### ISSUE TITLE
Support for async functions

### ISSUE THREAD

**Issue by jugglinmike** [jshint/jshint#2604]
Proposal: https://github.com/tc39/ecmascript-asyncawait

Advanced to stage 2 on July 30, 2015 https://github.com/tc39/ecma262

Depends on gh-2601

**Comment by bterlson** [comment]
:+1:

**Comment by lukeapage** [comment]
#2077 may be of use

**Comment by rDr4g0n** [comment]
I commented on #2601, but this seems like the right place to comment specifically for async/await.

async/await is now [finished](https://github.com/tc39/proposals/blob/e8c03544751b071b973064ef7d437a34ac6752cd/finished-proposals.md), [merged into](https://github.com/tc39/ecma262/pull/692) the ES2017 specification, and shipping in chrome and ff.

Maybe it's time to dust off #2077 ?

**Comment by Simobain** [comment]
Ok, so now that node 8 is live. Async/Await is officially supported. 
Will it be available soon on jsHint ?

Thanks,

**Comment by lolmaus** [comment]
I'll just leave this here: http://eslint.org/

**Comment by carldunham** [comment]
I'll just leave this here: https://github.com/eslint/eslint#what-about-experimental-features

**Comment by johansyd** [comment]
You just lost my interest in jshint. Sorry, to slow.

**Comment by SeregPie** [comment]
Node.js supports async/await. Every modern browser supports async/await. JSHint does not support it. This topic is 2 years old.

Thank you JSHint. It was good while it lasted.

Switched to ESLint.

**Comment by Grafluxe** [comment]
Hi team, 

The async spec has reached [stage 4](https://github.com/tc39/proposals/blob/master/finished-proposals.md), when can we expect to see support?

**Comment by tomaszkunicki** [comment]
RIP Jshint

**Comment by caitp** [comment]
That is not a helpful comment. Please submit a patch, rather than being negative.

**Comment by Grafluxe** [comment]
I was a fan of jshint, but have now decided to move to eslint. The goal of ECMAs new release cycle — and the creation of transpilers — is to allow people to use the lastest-greatest features of JavaScript early. Async/await has been at stage 4 for over a month now (and baked into Node and many mainstream browsers for longer), but jshint has yet to add support.

I used jshint daily, so thank you for creating something that was more flexible than jslint. I now need a more powerful and up-to-date tool with support for the latest ES features as well experimental ones. 

All the best.

**Comment by ddcovery** [comment]
We have hundred of thousands code lines validated with jshint.  All code uses Promises thoroughly.  6 months ago we decided to intruduce async/await to simplify the server side code (promises causes problems similar to callbacks when functions are not trivial: await will change naturally our code to a more readable/maintainable one).
The problem was, OMG, jshint.  Migrating to eslint is not trivial and we just decided to wait some time to JSHINT.

Thanks for the good job and for saving our code of errors this last 3 years, but we can't wait.

**Comment by jugglinmike** [comment]
We're hard at work on support for the next minor release. I'll remove the word "experimental" from the issue title to help avoid confusion.

**Comment by cikasfm** [comment]
+1 for this!

**Comment by oskarkrawczyk** [comment]
We're using the newest version of JSHint in @jsfiddle, thanks for the great tool! 

We got a few complaints about the linter complaining about `async` and `await`.

@jugglinmike since your comment was made 3 months ago, do you think this will get patched anytime soon?

**Comment by colxi** [comment]
Wich is the current status of async/await support ?

**Comment by tifDev** [comment]
Good luck with the async/await feature, its a must for this awesome plugin

**Comment by SuEric** [comment]
Any updates on this? I'm still getting error when used on anonymous functions

`Expected '(' and instead saw '{'.`

**Comment by DAmNRelentless** [comment]
Seems like we waited enough. Bye JSHint. It was a good time.

**Comment by nodesocket** [comment]
`async/await` are now supported as of version `2.10.1`. Simply update your `.jshintrc` to use `"esversion": 9`.

See the [CHANGELOG.md](https://github.com/jshint/jshint/blob/master/CHANGELOG.md).

**Comment by patrickdark** [comment]
@nodesocket The `esversion: 8` flag is apparently sufficient for `async` and `await` (though, as the change log states, there _is_ an `esversion: 9` flag).

In any case; thank you to the people that finally made this happen.

**Comment by jugglinmike** [comment]
@nodesocket's got it right--support for async functions can be enabled in JSHint as o 2.10.0. Closing this issue as resolved.
